### PR TITLE
Configure fixer if RuleSet config for it is <null>

### PR DESCRIPTION
**Pull request by SpacePossum**
ping @kubawerlos 

there is also a failing test I've to figure out but doesn't seem like a lot of trouble

### CODE REVIEW

File: UPGRADE-v3.md [hunk 1]
```diff
@@ -76,3 +76,10 @@ Rule | Option | Old value | New value
 Rule set | Note
 -------- | ----
 `@PHP56Migration` | was empty
+
+Changes to FixerFactory
+-----------------------
+
+The FixerFactory will now configure configurable fixers with the default config value (`[]`) when the rule
+set configuration for such fixer is set to `null`. Previously the fixers method `::configure` would not be called
```
Comment (kubawerlos) [inline UPGRADE-v3.md]: > Previously the fixers method `::configure` would not be called

Previously it was called in https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.16.1/src/AbstractFixer.php#L55.

**Comment by kubawerlos** [comment]
@SpacePossum I was thinking of moving the call from https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.16.1/src/AbstractFixer.php#L55 to the factory in the current major as well, it is called anyways, so won't be any BC, right?

File: UPGRADE-v3.md (hunk 1 above)
Comment (SpacePossum) [reply UPGRADE-v3.md]: check, I was referring to https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.16/src/FixerFactory.php#L178

**Comment by SpacePossum** [comment]
Should a fixer configure itself with default configuration during construction though? I think it is not the responsibility of the fixer and whatever is using the fixer must configure it (through `configure`) before calling the fixers `fix` - method with the appropriate configuration. WDYT?

File: src/FixerFactory.php [hunk 2]
```diff
@@ -179,6 +179,8 @@ public function useRuleSet(RuleSetInterface $ruleSet)
                 } else {
                     throw new InvalidFixerConfigurationException($fixer->getName(), 'Is not configurable.');
                 }
+            } elseif ($fixer instanceof ConfigurableFixerInterface) {
+                $fixer->configure([]);
```
Comment (keradus) [inline src/FixerFactory.php:183]: this should not be needed

this means someone would also need to do this:
```php
$abc = new AbcFixer();
$abc->configure([]);
```
it should be enough to run constructor to have fixer ready for use

**Pull request closed without merging**